conda/lock-environment.sh

```diff
@@ -25,7 +25,7 @@ fi
 
 # Local environments
 ## Generate explicit lock files
-conda-lock lock --mamba -f ${ENV_FILE}
+conda-lock lock -f ${ENV_FILE} -p "${PLATFORM}"
```

Comment:
@tsnow03 this explicitly sets the locking platforms _only_ to the Linux platform by default; I don't use conda-lock enough to know which platforms the old v1 conda-lock locked by default.